### PR TITLE
Move util + add tests

### PR DESCRIPTION
**Pull request by yaron1m**

### CODE REVIEW

File: packages/functions/src/donor/CancelAppointmentHandler.ts [hunk 1]
```diff
@@ -34,7 +34,8 @@ export default async function (
     console.error("Error notifying on cancelled appointment", appointment.id, e)
   );
 
-  const updatedAppointment = removeDonorFromDbAppointment(appointment);
+  const updatedAppointment =
+    DbAppointmentUtils.removeDonorFromDbAppointment(appointment);
```
Comment (evbambly) [inline packages/functions/src/donor/CancelAppointmentHandler.ts:38]: Doesn't this negate the concept of working with a DAL?

**Review by evbambly — APPROVED** [review]

File: packages/functions/src/donor/CancelAppointmentHandler.ts (hunk 1 above)
Comment (yaron1m) [reply packages/functions/src/donor/CancelAppointmentHandler.ts:38]: I don't understand why this should be in the DAL.

It is not reading or writing any data, just mutating an object.

This is a classic util.

What do you think? Should this be in the DAL? Why?

Comment (evbambly) [reply packages/functions/src/donor/CancelAppointmentHandler.ts:38]: Mutating is CRUD, I understand DAL === CRUD?

If there should be any complex permission handling for specific mutating operations,
they should be handled directly and not in a util file, no?

Comment (yaron1m) [reply packages/functions/src/donor/CancelAppointmentHandler.ts:38]: > Mutating is CRUD, I understand DAL === CRUD?

🤷 

> If there should be any complex permission handling

What permissions handling?

**Pull request merged**